app/Models/Beatmapset.php

```diff
@@ -621,9 +628,30 @@ public function setApproved($state, $user, ?array $beatmapIds = null)
             $this->previous_queue_duration = ($this->queued_at ?? $this->approved_date)->diffinSeconds();
             $this->queued_at = null;
         } elseif ($this->isPending() && $state === 'qualified') {
-            $maxAdjustment = ($GLOBALS['cfg']['osu']['beatmapset']['minimum_days_for_rank'] - 1) * 24 * 3600;
-            $adjustment = min($this->previous_queue_duration, $maxAdjustment);
-            $this->queued_at = $currentTime->copy()->subSeconds($adjustment);
+            // Check if any beatmaps where added after most recent invalidated nomination.
+            $lastResetNominationTime = $this->beatmapsetNominations()->current(false)->max('reset_at');
+            if (
+                $lastResetNominationTime !== null
+                && $this->beatmaps()->where('last_update', '>', $lastResetNominationTime)->exists()
```

Comment:
doesn't updating beatmap also update this column?

Reply:
Indeed it does 🤔

Reply:
should deleting a map reset the time?

Reply:
The reason for resetting on new difficulties is so they don't skip the queue when they could be introducing new problems and end up needing last minute popping/unrank or whatever; deleting won't be introducing new issues in the map set.